### ISSUE TITLE
Input.Placeholder in Input with component="button" not centered

### ISSUE THREAD

**Issue by quertc** [mantinedev/mantine#3352]
### What package has an issue

@mantine/core

### Describe the bug

By default, `Input.Placeholder` is slightly lower than necessary and differs from the native placeholder in `Input`. However, in the sandbox I have not been able to reproduce this (I use mantine in tauri), what could this problem be?

### What version of @mantine/hooks page do you have in package.json?

^5.10.0

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

### Possible fix

If we add such styles to `Input`, the problem will be solved:

```tsx
styles={theme => ({
   input: {
     lineHeight: theme.lineHeight,
   },
})}
```

With `lineHeight: 'normal'` the result is the same, except if the `Input` has `size="sm"`, this also applies to the `FileInput` component.

**Comment by haru-Blank** [comment]
![image](https://user-images.githubusercontent.com/29826638/218418752-ba2a4ae8-b553-428c-8ddd-11b005128f5a.png)
```javascript
<Input component="p"
    <Input.Placeholder>Hello placceholder</Input.Placeholder>
</Input>
``` 
or `<TextInput placeholder="..."/>`
The placeholder doesn't seem to vertically centered .
Can I work on it and try to fix it? @rtivital
There seem to be mismatch between `height` and `line-height` so it's not center.

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake and it is still an issue in version 7.0, please reopen it.
